### ISSUE TITLE
Unable to connect to server

### ISSUE THREAD

**Issue by ajnozari** [TheRemote/MinecraftBedrockServer#130]
I have a server that was working previously (before 1.19.20).  My client updated today but now I'm suddenly unable to connect?

Not sure if something changed but I even tried re-creating the world entirely (fresh install) and I still cannot connect.

**Comment by TheRemote** [comment]
Hey ajnozari,

Nothing has really changed no.  I'm surprised you're getting it with a completely fresh install.  There have been a few people that have had crashes after some of the upgrades but these all turned out to be corrupted servers.

Yours seems like a different issue for sure.  What is your host OS?  Is it running something like Ubuntu 20.04 or older?  If so it may just getting too old to run the latest dependencies that Microsoft is linking to.  The reason this came to mind is that when people's break out of the blue with really strange issues like this it has been that their OS was aging and needed an upgrade.

At the very least there's no recent changes that could explain this.  I only recently added a clean.sh script that basically forces it to download and install the latest server version.  It could be related to updates to the Minecraft binary itself like what Microsoft is linking to.

You could try running the old version with the version override / pin_version.txt theoretically if you remember the exact old version.  It would be interesting if this solves it.  I wouldn't necessarily expect this but it's possible if Microsoft changed dependencies and this is not the latest version of your OS.

I take it you are seeing nothing when trying to connect in either the server console or on the client or you probably would have mentioned it.  Some more information about the environment may help if none of this helps solve it.  I'm not totally sure about this one yet but wanted to let you know what I could to help aid with investigating this a little at least!

**Comment by ajnozari** [comment]
As an update I have upgraded to v. 22.04 and I'm still having issues.  I have the server backed up, I'm going to re-install the OS entirely and see if that helps.

**Comment by TheRemote** [comment]
Hey ajnozari,

Check the firewall for sure.  Most people who have been having issues on Ubuntu either have ufw (universal firewall) or firewalld installed.  These are *not* installed by default on Ubuntu BUT people who use VMs often have them enabled anyway.

Is it Oracle?  If it's Oracle I can tell you the exact answer because that one comes up all the time:  https://jamesachambers.com/official-minecraft-bedrock-dedicated-server-on-raspberry-pi/#comment-13946

Basically the only explanation here is firewall (probably enabled without your knowledge by the VM provider).

`ufw allow 19132`

or 

```
sudo firewall-cmd –permanent –zone=public –add-port=19132/tcp
sudo firewall-cmd –permanent –zone=public –add-port=19132/udp
sudo firewall-cmd –reload
```

will probably do it.

**Comment by ajnozari** [comment]
The UFW is inactive that's what is weird.  That was definitely the first thing I checked.

The logs also don't show anything special, it's like the entire VM just isn't responding at all.

**Comment by TheRemote** [comment]
You can use wireshark and do a packet trace if you'd like.  I really doubt traffic is flowing.  We've troubleshooted a Hyper-V VM and found out it had a faulty virtual adapter once and they had to change the VM network adapter type.  

I've only done it one time but you can install wireshark if you're positive that traffic is flowing and that may reveal a different issue.  It did in that case!

**Comment by ajnozari** [comment]
This is Hyper-v.  I know that older versions of ubuntu used legacy network adapters but i'm on server 2019 so I only have one version available and it was 100% working before.

**Comment by TheRemote** [comment]
Oh boy.  I was afraid you were going to say it was Hyper-V!  Okay, that is literally the one that we did the only Wireshark trace I've ever done troubleshooting this.  

It's literally the weirdest issue I've ever seen and yours is a 100% match.  I bet if you install wireshark you will see the *exact* same thing we found before.  Here's a link:  https://jamesachambers.com/minecraft-bedrock-edition-ubuntu-dedicated-server-guide/#comment-13863

The answer ended up being this:

```
I switched the VM to back to using Generation 1, and selected the legacy lan driver in hyper-v, it’s working now.

Must be a bug of some kind in the Linux Hyper-v lan driver for the Generation 2 vm spec.
```

However your message concerns me a little bit.  Is this not possible for you?  Does 2019 not let you create a Gen1 VM anymore?

**Comment by ajnozari** [comment]
It does, however the weird thing is, Ubuntu natively supports the gen2.

I'll try making a new gen 1 and give it a test.

**Comment by ajnozari** [comment]
This must be a new bug since everything was working as of like a week ago.  Thanks MSFT

**Comment by TheRemote** [comment]
It could have to do with how they are linking to the libraries.  They seem to be switching between libssl1.1 and libssl3 (right now the server links to both libraries).

It is pretty interesting that yours was working before and broke.  I couldn't explain that other than it was definitely something Microsoft did and that there's no explanation within the scripts that I've done at least that would make any sense to explain this.

I'm glad you got yours going, enjoy and take care, let me know if I can help further!
